docs/docs/pyproject.md

```diff
@@ -192,6 +192,15 @@ poetry = 'poetry:console.run'
 
 Here, we will have the `poetry` script installed which will execute `console.run` in the `poetry` package.
 
+To specify a script that
+[depends on an extra](https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies),
```

Comment:
I think it's preferable to point to the `extras` section of the Poetry documentation instead of `setuptools` since they are different build tools.